tests/cpp/test_cpp_client.py

```diff
@@ -38,16 +38,23 @@ def get_test_names():
     """Obtain test names by globbing for client_test
     Add tests manually if necessary
     """
-    glob_path = osp.join(TEST_PATH, "build/client_test*")
+    glob_path = osp.join(TEST_PATH, "client_test*")
     test_names = glob(glob_path)
+    test_names = list(filter(lambda test: test.find('.h') == -1, test_names))
     test_names = [(pytest.param(test,
                                 id=osp.basename(test))) for test in test_names]
     return test_names
 
 @pytest.mark.parametrize("test", get_test_names())
-def test_cpp_client(test, use_cluster):
-    cmd = []
-    cmd.append(test)
+def test_cpp_client(test, use_cluster, build):
+    # Build the path to the test executable from the source file name
+    # . keep only the last three parts of the path: (tests, language, basename)
+    test = "/".join(test.split("/")[-3:])
```

Comment:
Same as above about rewriting using pathlib

Reply:
No clean way to do thus via pathlib since we have to keep three levels of path